pynetdicom/transport.py

```diff
@@ -384,52 +217,32 @@ def tls_args(self, tls_args):
         self._tls_args = tls_args
 
 
-class RequestHandler(BaseRequestHandler):
-    """Connection request handler for the ``AssociationServer``.
+class AssociationProtocol(asyncio.Protocol):
+    def __init__(self, active_connections=[]):
+        self._assoc = None
+        self._stream = None
+        self._active_connections = active_connections
 
-    .. versionadded:: 1.2
-
-    Attributes
-    ----------
-    client_address : 2-tuple
-        The ``(host, port)`` of the remote.
-    request : socket.socket
-        The (unaccepted) client socket.
-    server : transport.AssociationServer or transport.ThreadedAssociationServer
-        The server that received the connection request.
-    """
-    @property
-    def ae(self):
-        """Return the server's parent AE."""
-        return self.server.ae
-
-    def handle(self):
+    def connection_made(self, transport):
         """Handle an association request.
 
         * Creates a new Association acceptor instance and configures it.
         * Sets the Association's socket to the request's socket.
         * Starts the Association reactor.
         """
-        assoc = self._create_association()
+        self._assoc, self._stream = self._create_association(transport)
 
         # Trigger must be after binding the events
         evt.trigger(
-            assoc, evt.EVT_CONN_OPEN, {'address' : self.client_address}
+            self._assoc, evt.EVT_CONN_OPEN, {'address' : self.client_address}
         )
 
-        assoc.start()
+        self._active_connections.append(self)
 
-    @property
-    def local(self):
-        """Return a 2-tuple of the local server's ``(host, port)`` address."""
-        return self.server.server_address
+        asyncio.create_task(self._assoc.start())
```

Comment:
note how association reactor is a asyncio task. This is inspired by line in asyncio.streams here: https://github.com/python/cpython/blob/89e50ab36fac6a0e7f1998501f36fcd2872a6604/Lib/asyncio/streams.py#L244